src/cgutils.cpp

```diff
@@ -1943,8 +1941,10 @@ static void emit_memcpy_llvm(jl_codectx_t &ctx, Value *dst, MDNode *tbaa_dst, Va
         // machine size vectors this should be enough.
         const DataLayout &DL = jl_Module->getDataLayout();
         auto srcty = cast<PointerType>(src->getType());
+        //TODO unsafe nonopaque pointer
         auto srcel = srcty->getElementType();
         auto dstty = cast<PointerType>(dst->getType());
+        //TODO unsafe nonopaque pointer
```

Comment:
We could probably convert this soon (in a separate PR) to use our own idea of a preferred copy size, based on `align` and `sz`. Either as an `[ align x n ]` (array of align elements) or `iSZ` element (integer of sz bytes) if a power of two.

Reply:
So it looks like the original justification for this block of inlined memcpy code is #23352, from ~4.5 years ago. Since the reasoning given there was an unnecessary bitcast being inserted which caused loop-vectorize to fail to vectorize a loop, I'm kind of tempted to just delete this block of code altogether and see if either 4.5 years of LLVM updates or the transition to opaque pointers will make this memcpy inlining unnecessary. If there's a reproducer for the old bug, I can check it out on a local branch to see if it's still an issue.